src/main/kotlin/no/item/gradle/GenerateTypeScriptTask.kt

```diff
@@ -0,0 +1,29 @@
+package no.item.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import javax.inject.Inject
+
+open class GenerateTypeScriptTask @Inject constructor(private val extension: GenerateTypeScriptExtension) : DefaultTask(){
+
+    @TaskAction
+    fun generateTypeScript() {
+      val files = findXmlFiles();
+      println("Found XML files: ")
+      println("--------------------")
+      files.forEach { println(it) }
+      //2.send dem til en parser
+      //3.send resultatet til en writer
+      //4.exit
+    }
+
+  fun findXmlFiles(): Sequence<String> {
+    val files = File(extension.fileDir)
```

Comment:
I would return this directly without setting it to `val files` first.

Reply:
I agree.